src/main/java/org/commcare/suite/model/PostRequest.java

```diff
@@ -48,7 +48,12 @@ public URL getUrl() {
     public Multimap<String, String> getEvaluatedParams(EvaluationContext evalContext) {
         Multimap<String, String> evaluatedParams = ArrayListMultimap.create();
         for (QueryData queryData : params) {
-            evaluatedParams.putAll(queryData.getKey(), queryData.getValues(evalContext));
+            Iterable<String> val = queryData.getValues(evalContext);
+            if (val.iterator().hasNext()) {
+                evaluatedParams.putAll(queryData.getKey(), val);
+            } else {
+                evaluatedParams.put(queryData.getKey(), "");
```

Comment:
This is changing the value that get's put in right? Before if there were no values then the key would not be present in `evaluatedParams` but now the key will be present with an empty string as the value. Making that change will also affect the actual POST request and I think it will require changes on HQ's side.

I think there are 2 options to avoid this:
1. Pass in a boolean arg to determine if blank keys should be included
    - This is simpler but may not be the best long term option if we end up needing to support empty strings as post param values (maybe that's unlikely).
3. Use a virtual instance which would then support using the `count` function to determine if the key was present.
    - More complex and would require changing the 'relevant' expression but is the most flexible and future proof.

Reply:
Ahh right! I do think it's quite unlikely that we need to support blank values in post params for now so I am inclined towards the simpler option 1 here.

Reply:
f26f65dd60129d2cddb3d8a21f67944064b5cff2